tests/test_blacklisting.py

```diff
@@ -118,7 +120,7 @@ def test_federation_client_allowed_ip(self, resolver):
             )
         ]
 
-        request.render(self.sydent.servlets.registerServlet)
+        request.render(self.register_resource)
```

Comment:
I wonder if you could use the actual resources created within `self.sydent` rather than creating a new instance.

```suggestion
        resource = self.sydent.clientApiHttpServer.factory.getResourceFor(request)
        request.render(resource)
```

possibly.  Or even:

```python
channel.site = self.sydent.clientApiHttpServer.factory
request.process()
```

Reply:
Bah, this is rather annoying. It seems sydent doesn't quite have as good of a setup for testing as synapse. Still working through it though.

Reply:
Done in 7a78a050c5c9eb8ecd57026949718ad1e1744f7e, `process()` actually gets called deep internal in Twisted so it didn't quite end up as you suggested, but I think I did the gist of what you wanted.